src/app/modules/item/services/item-task.service.ts

```diff
@@ -0,0 +1,122 @@
+import { FullItemRoute, itemRoute } from 'src/app/shared/routing/item-route';
+import { ItemRouter } from 'src/app/shared/routing/item-router';
+import { Task, TaskPlatform } from '../task-communication/task-proxy';
+import { TaskParamsValue } from '../task-communication/types';
+import { ItemTaskAnswerService } from './item-task-answer.service';
+import { ItemTaskInitService } from './item-task-init.service';
+import { ItemTaskViewsService } from './item-task-views.service';
+
+@Injectable()
+export class ItemTaskService {
+  task$ = this.initService.task$;
+  iframeSrc$ = this.initService.iframeSrc$;
+  get initialized(): boolean {
+    return this.initService.initialized;
+  }
+
+  views$ = this.viewsService.views$;
+  display$ = this.viewsService.display$;
+  activeView$ = this.viewsService.activeView$;
+
+  private config$ = this.initService.config$;
+
+  constructor(
+    private initService: ItemTaskInitService,
+    private answerService: ItemTaskAnswerService,
+    private viewsService: ItemTaskViewsService,
+    private itemRouter: ItemRouter,
+    private itemNavigationService: ItemNavigationService,
+    private router: Router,
+    private localeService: LocaleService,
+  ) {}
+
+  configure(route: FullItemRoute, url: string, attemptId: string): void {
+    this.initService.configure(route, url, attemptId);
+  }
+
+  initTask(iframe: HTMLIFrameElement): void {
+    this.initService.initTask(iframe, task => this.bindPlatform(task));
+  }
+
+  showView(view: string): void {
+    this.viewsService.showView(view);
+  }
+
+  private bindPlatform(task: Task): void {
+    const platform = new TaskPlatform({
+      validate: (mode): Observable<void> => this.validate(mode).pipe(mapTo(undefined)),
+      getTaskParams: (): Observable<TaskParamsValue> =>
+        of({ minScore: -3, maxScore: 10, randomSeed: 0, noScore: 0, readOnly: false, options: {} }),
+      updateDisplay: (display): Observable<void> => {
+        this.viewsService.updateDisplay(display);
+        return EMPTY;
+      },
+      showView: (view): Observable<void> => {
+        this.viewsService.showView(view);
+        return EMPTY;
+      },
+      openUrl: (params): Observable<void> => {
+        if (typeof params === 'string') return this.navigateToItem(params);
+        if ('path' in params) return this.navigateToItem(params.path, params.newTab);
+        this.navigate(params.url, params.newTab);
+        return EMPTY;
+      }
+    });
+    task.bindPlatform(platform);
+  }
+
+  private validate(mode: string): Observable<unknown> {
+    switch (mode) {
+      case 'cancel': return this.answerService.reloadAnswer();
+      case 'nextImmediate': return this.navigateToNextItem();
+      case 'next': return this.answerService.submitAnswer().pipe(switchMap(() => this.navigateToNextItem()));
+      case 'top': return this.answerService.submitAnswer().pipe(switchMap(() => this.scrollTop()));
+      default: return this.answerService.submitAnswer();
```

Comment:
Is there a reason to be so "open" to any string and just submit the answer in case of random string? This function is "called" by the task?

Reply:
Maybe not, but I got it from here: https://github.com/France-ioi/AlgoreaPlatform/blob/master/task/taskDirective.js#L236

Apparently, unless the mode is `'cancel'` or `'nextImmediate'`, the answer is submitted
Plus it makes sense because when testing the validate function, the mode can have various values like `'done'` that expect validation, so I figured it was intended to work this way

Reply:
🤷‍♂️

Reply:
Does this "shrug" calls to any action ? :thinking: